packages/core/security/core-security-server/src/contracts.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { CoreAuthenticationService } from './authc';
+import type { CoreInternalSecurityContract } from './api_provider';
+
+/**
+ * Setup contract for Core's security service.
+ *
+ * @public
+ */
+export interface SecurityServiceSetup {
+  /**
+   * Register the security implementation that then will be used and re-exposed by Core.
+   *
+   * @remark this should **exclusively** be used by the security plugin.
```

Comment:
When we discussed about this, (iirc) we came to the conclusion that it was fine to not do any hard-checks, because
1. it's already possible to hook up into security by mimicking what the security plugin does today
2. the implementation remains in "xpack licensed" code
3. ideally we would be avoided doing hard checks on plugin ids (and also it isn't really protecting against 3rd party replacement as they could just the same pluginId as our internal security plugin...

I'll let @elastic/kibana-security confirms that it's still fine

Reply:
>I'll let @elastic/kibana-security confirms that it's still fine

Yeah, I believe it's fine to keep what we have now. There are so many hurdles and restrictions at various levels to replace the built-in Security plugin with any alternative implementation that it sounds impractical for anyone to even consider.